hera_cal/smooth_cal.py

```diff
@@ -298,16 +342,16 @@ def time_freq_2D_filter(gains, wgts, freqs, times, freq_scale=10.0, time_scale=1
             # Generate filters if not provided
             if design_matrix is None:
                 design_matrix = dpss_filters(
-                    xout[1], xout[0], freq_scale=freq_scale, time_scale=time_scale, eigenval_cutoff=eigenval_cutoff
+                    freqs=xout[1], times=xout[0], freq_scale=freq_scale, time_scale=time_scale,
+                    eigenval_cutoff=eigenval_cutoff
                 )
 
-            # Fit solution matrix if not provided
-            if sol_matrix is None:
-                sol_matrix = fit_solution_matrix(wout, design_matrix)
+            time_filters, freq_filters = design_matrix
 
-            # Filter gains
-            filtered = design_matrix @ (sol_matrix @ gout.reshape(-1))
-            filtered = filtered.reshape(gout.shape)
+            # Filter gain solutions
+            filtered, fit_info = solve_2D_DPSS(
+                gains=gout, weights=wout, time_filters=time_filters, freq_filters=freq_filters, XTXinv=sol_matrix
```

Comment:
Likewise, I think we should decide whether we want to call this the sol_matrix or XTXinv. Let's be consistent throughout the code. I wouldn't worry about breaking interfaces, your previous PR is too new.